### ISSUE TITLE
Adaptation to a "different" static graph

### ISSUE THREAD

**Issue by marcopost-it** [Lee-zix/RE-GCN#10]
Hi!

First of all, thank you for sharing the code for your publication.

I am working to adapt your framework to my problem, where "static" relations regard nodes which are already in the knowledge graph. As a consequence, when setting static_graph.ndata['h'] (rrgcn.py, line 146), I don't need to concatenate dynamic embeddings with word embeddings (which I have deleted), and thus when I take the output of the static layer I don't have to limit to the first self.num_ents elements.

However, with this settings, the entity-prediction loss is nan. I've tried to solve this problem by setting self.h as the sum of the dynamic and static embeddings (rrgcn.py, line 150 and 176). In this way, the problem with the entity-prediction loss is solved, but the static loss has very big values (>2000) and does not decreases over the training epochs. The result is that the static graph doesn't allow to improve performance.

Since this is a methodological question rather than a coding issue, I was wondering if you could help me with your feedback and guidance. Your ideas on how to adapt the methodology and code to work with my task would have a great value to me.

Thanks,
Marco

**Comment by Lee-zix** [comment]
Sorry for the delay due to my thesis defense!

Thanks for your interest. The reason for your first question, in my opinion, is that self-loop connection is moved in the R-GCN for the static graph. Thus, if there exists an isolated node in your static graph. DGL may cause the nan loss. But this is only my guesswork. Hope it helps.

As for the second question,  in my opinion, static graph constraint is an approach to engage more information. If the results can increase to the reported result of RE-GCN(without static graph) when you decrease the edges in your static graph, it shows that the codes are correct. However, I cannot give you more suggestions without knowing your static graph and your task situation.

**Comment by marcopost-it** [comment]
No worries for the delay. I'm sure that your defense has been great!

I managed to solve my problems. They were related to the fact that in my temporal graph I have patients and biomedical concepts, but obviously the static graph contains only relationships between biomedical concepts. I solved my problems by giving just the biomedical concepts to the static GCN layer. 

Thank you for your answer!
